### PR TITLE
Copy RandomUniform `m_state` during clone_with_new_inputs()

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - CVS-73450

### CODE REVIEW

File: src/core/src/op/random_uniform.cpp [hunk 1]
```diff
@@ -119,12 +119,16 @@ bool op::v8::RandomUniform::visit_attributes(AttributeVisitor& visitor) {
...
     check_new_args_count(this, new_args);
-    return make_shared<v8::RandomUniform>(new_args[0],
+    auto a = make_shared<v8::RandomUniform>(new_args[0],
                                           new_args[1],
                                           new_args[2],
                                           m_output_type,
                                           m_global_seed,
                                           m_op_seed);
+    // Copy RandomUniform state
+    std::lock_guard<std::mutex> guard(a->m_state_mutex);
```
Comment (ilyachur) [inline src/core/src/op/random_uniform.cpp:129]: Looks like you don't need this mutex.

**Review by ilyachur — COMMENTED** [review]
Please add new tests and fix code style

**Review by a-sidorova — APPROVED** [review]

File: src/core/src/op/random_uniform.cpp (hunk 1 above)
Comment (vurusovs) [reply src/core/src/op/random_uniform.cpp]: Thanks, done!

Comment (vurusovs) [reply src/core/src/op/random_uniform.cpp]: Test to be added soon

Comment (vurusovs) [reply src/core/src/op/random_uniform.cpp]: Tests should be added in scope of CVS-69723

Comment (vurusovs) [reply src/core/src/op/random_uniform.cpp]: Added unit test for `clone_with_new_inputs()`

**Review by popovaan — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**